### PR TITLE
sentinel: Add store methods to insert and query vulnerabilities

### PR DESCRIPTION
**Pull request by efritz**
Pulled from #47531. This PR adds the methods related to querying and inserting vulnerabilities. Will be populated by the CVE scanner background job and queried by the GraphQL layer in the future.

## Test plan

Included unit tests.

### CODE REVIEW

File: enterprise/internal/codeintel/sentinel/internal/store/store.go [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
 	logger "github.com/sourcegraph/log"
 
+	"github.com/sourcegraph/sourcegraph/enterprise/internal/codeintel/sentinel/shared"
 	"github.com/sourcegraph/sourcegraph/internal/database"
 	"github.com/sourcegraph/sourcegraph/internal/database/basestore"
 	"github.com/sourcegraph/sourcegraph/internal/observation"
 )
 
 type Store interface {
+	VulnerabilityByID(ctx context.Context, id int) (_ shared.Vulnerability, _ bool, err error)
```
Comment (Numbers88s) [inline enterprise/internal/codeintel/sentinel/internal/store/store.go:15]: `GetVulnerabilityByID` perhaps? the pattern is off by one 😄

File: enterprise/internal/codeintel/sentinel/internal/store/vulnerabilities.go [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+			"data_source",
+			"urls",
+			"severity",
+			"cvss_vector",
+			"cvss_score",
+			"published",
+			"modified",
+			"withdrawn",
+		},
+		func(inserter *batch.Inserter) error {
```
Comment (Numbers88s) [inline enterprise/internal/codeintel/sentinel/internal/store/vulnerabilities.go:253]: I like the pattern where you move it to an inserter variable you pass in. Less nesting _shrug_

**Review by Numbers88s — COMMENTED** [review]
Looks good, I'd approve it but ill wait for when its done

File: enterprise/internal/codeintel/sentinel/internal/store/store.go (hunk 1 above)
Comment (efritz) [reply enterprise/internal/codeintel/sentinel/internal/store/store.go:15]: This is actually a weird convention for GraphQL/Relay IDs -> objects. I'd like to standardize it across all our services actually, but might wait post-sentinel.

Comment (Numbers88s) [reply enterprise/internal/codeintel/sentinel/internal/store/store.go:15]: Like it needs the store to be called that way? A bit odd, but could the store have the CRUD verb pattern and the service which is what would probably be called from the transform layer have the weird convention? Or maybe the convention is pushed to the transform layer too.

Comment (efritz) [reply enterprise/internal/codeintel/sentinel/internal/store/store.go:15]: It is _currently_, so doing a mechanical cleanup would be easy between job fair projects.

Comment (Numbers88s) [reply enterprise/internal/codeintel/sentinel/internal/store/store.go:15]: Yeah not blocking or suggesting right now. Just noticed it, thought I mention it 👍

**Review by Numbers88s — APPROVED** [review]

**Pull request merged**